systemd/systemd.go

```diff
@@ -1542,11 +1543,8 @@ func (s *systemd) EnsureMountUnitFileWithOptions(unitOptions *MountUnitOptions)
 		if err := s.EnableNoReload(units); err != nil {
 			return "", err
 		}
-		// In the case of mountCreated, ReloadOrRestart
-		// has the same effect as just Start.
-		// In the case of MountUpdate, we need to reload
-		// the unit.
-		if err := s.ReloadOrRestart(units); err != nil {
+		// Start/restart the created or modified unit now
+		if err := s.RestartNoWaitForStop(units); err != nil {
```

Comment:
hm wonder what systemd did here really, the mount units seem to have `CanReload=yes`, but it's not clear what this property means as it's not documented at all, other than 
https://github.com/systemd/systemd/blob/da95494ca3857c42d55ada987d50c254478d185c/src/core/job.h#L60
which wold be applicable to units that have this property. So indeed, if the unit file is changed, but the unit is already active I think this would not do anything useful, and thus an explicit restart is needed.

Reply:
Obviously if a revision was to be included in the mount unit name, then the problem would go away. In any way, it'd be useful to leave a comment in the code.

Reply:
I only see CanReload in https://www.freedesktop.org/software/systemd/man/latest/org.freedesktop.systemd1.html , but not sure if relevant here.

Regarding the comment, that would be in the code creating the mount unit instead of here?